rust/headless-client/src/standalone.rs

```diff
@@ -238,10 +237,19 @@ pub fn run_only_headless_client() -> Result<()> {
                 InternalServerMsg::Ipc(IpcServerMsg::OnDisconnect {
                     error_msg,
                     is_authentication_error: _,
-                }) => return Err(anyhow!(error_msg).context("Firezone disconnected")),
+                }) => break Err(anyhow!(error_msg).context("Firezone disconnected")),
                 InternalServerMsg::Ipc(IpcServerMsg::OnUpdateResources(_)) => {
-                    // On every resources update, flush DNS to mitigate <https://github.com/firezone/firezone/issues/5052>
+                    // On every Resources update, flush DNS to mitigate <https://github.com/firezone/firezone/issues/5052>
                     dns_controller.flush()?;
+                    if let Some(instant) = last_connlib_start_instant.take() {
+                        // `OnUpdateResources` appears to be the latest callback that happens during startup
```

Comment:
Technically, a user may not have any resources. What is guaranteed to happen is `on_update_interface_config`. Once we set the IPs on TUN, we are ready to route traffic. I think that may be the better place to put that.

Reply:
If the user sends a DNS query after `on_update_interface_config`, is it possible that connlib doesn't know the DNS resources yet?

Reply:
No, we receive them in the same message from the portal. Tunnel IPs, resources and relays are all updated atomically.

Reply:
Switching to `on_update_interface_config` then

idk if it would work for the other Clients but it would be cool to have them in the same callback / event here then, I assumed they might happen at different times

Reply:
Yeah @conectado and I talked about changing this today. Specifically though, it is the routes that need to be merged into this. The resource change separately.

Reply:
Okay. As long as I can treat one of them as "The nanosecond I get this, a user packet can hit the interface and do something productive" :)

Reply:
That is the interface update!